code/modules/mob/living/living.dm

```diff
@@ -465,8 +464,11 @@
 	if (s_active && !(CanReach(s_active,view_only = TRUE)))
 		s_active.close(src)
 
-	if(lying && !buckled && prob(getBruteLoss()*200/maxHealth))
-		makeTrail(newloc, T, old_direction)
+	on_drag(newloc, old_direction)
+
+/mob/living/proc/on_drag(newloc, dir)
+	if(prob(getBruteLoss()*200/maxHealth) && lying && !buckled && isturf(loc) && has_gravity())
+		makeTrail(newloc, loc, dir)
```

Comment:
you need the old turf or you'll break trails, loc will always be the same as newloc here